### PR TITLE
Fixed ignored region in language filter

### PR DESCRIPTION
**Pull request by Bullettoothtobi**
fixed and tested the region

https://github.com/sonata-project/SonataIntlBundle/issues/112

### CODE REVIEW

File: Templating/Helper/LocaleHelper.php [hunk 1]
```diff
@@ -43,7 +43,7 @@ public function language($code, $locale = null)
     {
         $codes = explode('_', $code);
 
-        $name = Intl::getLanguageBundle()->getLanguageName($codes[0], isset($codes[1]) ? $code[1] : null, $locale ?: $this->localeDetector->getLocale());
+        $name = Intl::getLanguageBundle()->getLanguageName($codes[0], isset($codes[1]) ? $codes[1] : null, $locale ?: $this->localeDetector->getLocale());
```
Comment (greg0ire) [inline Templating/Helper/LocaleHelper.php]: Can you linebreak this line while you're at it ? It is way too long.

File: Tests/Helper/LocaleHelperTest.php [hunk 2]
```diff
@@ -35,6 +35,7 @@ public function testLanguage()
         $helper = $this->getHelper();
         $this->assertEquals('français', $helper->language('fr'));
         $this->assertEquals('français', $helper->language('fr_FR'));
+        $this->assertEquals('anglais américain', $helper->language('en_US'));
```
Comment (greg0ire) [inline Tests/Helper/LocaleHelperTest.php]: Isn't line 37 already testing the piece of code you changed ? I don' t think it is possible to get a test that fails without your change, actually.

File: Templating/Helper/LocaleHelper.php [hunk 3]
```diff
@@ -43,7 +43,8 @@ public function language($code, $locale = null)
     {
         $codes = explode('_', $code);
 
-        $name = Intl::getLanguageBundle()->getLanguageName($codes[0], isset($codes[1]) ? $code[1] : null, $locale ?: $this->localeDetector->getLocale());
+        $name = Intl::getLanguageBundle()->getLanguageName($codes[0], isset($codes[1]) ? $codes[1] : null,
```
Comment (greg0ire) [inline Templating/Helper/LocaleHelper.php]: `Intl::getLanguageBundle()` is hardcoded in the test, that's why it is not possible to test if `getLanguageName` is called with the right thing…

File: Templating/Helper/LocaleHelper.php [hunk 4]
```diff
@@ -43,7 +43,8 @@ public function language($code, $locale = null)
     {
         $codes = explode('_', $code);
 
-        $name = Intl::getLanguageBundle()->getLanguageName($codes[0], isset($codes[1]) ? $code[1] : null, $locale ?: $this->localeDetector->getLocale());
+        $name = Intl::getLanguageBundle()->getLanguageName($codes[0], isset($codes[1]) ? $codes[1] : null, 
+            $locale ?: $this->localeDetector->getLocale());
```
Comment (greg0ire) [inline Templating/Helper/LocaleHelper.php]: Not sure this will pass cs validation. That's not how you're supposed to write a multiline call… here is what it should look like : 

```php
$name = Intl::getLanguageBundle()->getLanguageName(
    $codes[0],
    isset($codes[1]) ? $codes[1] : null,
    $locale ?: $this->localeDetector->getLocale()
);
```

File: Tests/Helper/LocaleHelperTest.php (hunk 2 above)
Comment (Bullettoothtobi) [inline Tests/Helper/LocaleHelperTest.php]: yes, but "fr" and "fr_FR" always give the same result. This is why I tested with "en_US", because it gives a different result than "en".

Comment (greg0ire) [inline Tests/Helper/LocaleHelperTest.php]: Oh ok. Good then, keep it.

**Comment by Soullivaneuh** [comment]
Please fix CS: https://travis-ci.org/sonata-project/SonataIntlBundle/jobs/126328914#L640

File: Templating/Helper/LocaleHelper.php (hunk 4 above)
Comment (Soullivaneuh) [inline Templating/Helper/LocaleHelper.php]: BTW, this looks greater to read.

**Comment by Bullettoothtobi** [comment]
I'm on it

File: Templating/Helper/LocaleHelper.php (hunk 3 above)
Comment (greg0ire) [inline Templating/Helper/LocaleHelper.php]: Someone should make another PR to solve this with dependency injection. I'm saying another PR because it will most likely be a BC-break.

**Comment by Soullivaneuh** [comment]
Thanks @Bullettoothtobi
